exchange/utils.go

```diff
@@ -847,6 +859,49 @@ func mergeBidderRequests(allBidderRequests []BidderRequest, bidderNameToBidderRe
 	return allBidderRequests
 }
 
+func gdprFromGPP(r *openrtb2.BidRequest, gpp gpplib.GppContainer) {
+	if r.Regs != nil && r.Regs.GDPR == nil {
+		if len(r.Regs.GPPSID) > 0 {
+			// Set to 0 unless SID exists
+			r.Regs.GDPR = ptrutil.ToPtr[int8](0)
+			for _, id := range r.Regs.GPPSID {
+				if id == int8(gppConstants.SectionTCFEU2) {
+					r.Regs.GDPR = ptrutil.ToPtr[int8](1)
```

Comment:
I believe r.Regs and r.User is shared memory at this point in code, right? We will need to make a copy of these objects before applying a bidder specific setting. Are there tests we can add a higher level that would catch this scenario?

Reply:
OH, yes, that is a good catch. I will have to think about how it may be possible to test for. Can certainly rig a test to catch this one specific instance. But a test to generally catch that some conditional code in PBS Core might overwrite shared memory would be difficult, as it would require a test that covers all possible conditions and then checks for changes in shared memory. Hard to cover for conditions that haven't been written yet.

Reply:
No worries if it's too hard to instrument. Thank you for investigating.